test/org/opensolaris/opengrok/configuration/messages/RepositoryMessageTest.java

```diff
@@ -0,0 +1,144 @@
+/*
+ * CDDL HEADER START
+ *
+ * The contents of this file are subject to the terms of the
+ * Common Development and Distribution License (the "License").
+ * You may not use this file except in compliance with the License.
+ *
+ * See LICENSE.txt included in this distribution for the specific
+ * language governing permissions and limitations under the License.
+ *
+ * When distributing Covered Code, include this CDDL HEADER in each
+ * file and include the License file at LICENSE.txt.
+ * If applicable, add the following below this CDDL HEADER, with the
+ * fields enclosed by brackets "[]" replaced with your own identifying
+ * information: Portions Copyright [yyyy] [name of copyright owner]
+ *
+ * CDDL HEADER END
+ */
+
+/*
+ * Copyright (c) 2018, Oracle and/or its affiliates. All rights reserved.
+ */
+package org.opensolaris.opengrok.configuration.messages;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.concurrent.ConcurrentHashMap;
+import org.junit.After;
+import org.junit.Assert;
+import org.junit.Assume;
+import org.junit.Before;
+import org.junit.Test;
+import org.opensolaris.opengrok.condition.ConditionalRun;
+import org.opensolaris.opengrok.condition.RepositoryInstalled;
+import org.opensolaris.opengrok.configuration.RuntimeEnvironment;
+import org.opensolaris.opengrok.history.GitRepository;
+import org.opensolaris.opengrok.history.HistoryGuru;
+import org.opensolaris.opengrok.history.MercurialRepository;
+import org.opensolaris.opengrok.history.MercurialRepositoryTest;
+import org.opensolaris.opengrok.history.RepositoryFactory;
+import org.opensolaris.opengrok.history.RepositoryInfo;
+import org.opensolaris.opengrok.history.SubversionRepository;
+import org.opensolaris.opengrok.index.Indexer;
+import org.opensolaris.opengrok.util.TestRepository;
+
+/**
+ * Test repository message handling.
+ * 
+ * @author Vladimir Kotal
+ */
+public class RepositoryMessageTest {
+    
+    RuntimeEnvironment env;
+
+    private static TestRepository repository = new TestRepository();
+    
+    @Before
+    public void setUp() throws IOException {
+        Assume.assumeTrue(new MercurialRepository().isWorking());
+        Assume.assumeTrue(new SubversionRepository().isWorking());
+        Assume.assumeTrue(new GitRepository().isWorking());
```

Comment:
I think I included the repeateable conditional rule for this, can you try it?

Reply:
these are not needed, removed.